tools/depmod.c

```diff
@@ -2855,7 +2855,7 @@ static int depfile_up_to_date_dir(DIR *d, time_t mtime, size_t baselen, char *pa
 /* uptodate: 1, outdated: 0, errors < 0 */
 static int depfile_up_to_date(const char *dirname)
```

Comment:
Proper filesystem handling is always hard, and the best solution would be a proper API in POSIX, but that would be even more day-dreaming.

The realpath(3) approach sounds good at first, but keep in mind that PATH_MAX has to be seen "relative." There is nothing wrong in having a directory structure which far exceeds PATH_MAX characters in "real path" as long as I can perform a relative movement from current directory to target directory within PATH_MAX characters limits.

I'm not sure if it's relevant for kmod project though, given that so many projects simply stop working properly if a path exceeds 1000 or PATH_MAX or any other chosen limit.

Reply:
for real use cases that we support arbitrarily long strings, we use strbuf since it allows maintaining the common case in the stack and allocates/free for the exceptional cases. strbuf was a very late addition in the project though, so it's not used as much as it should. Fix here LGTM.

Reply:
@evelikov  if the function is only ever expected to receive an array of a given size, using `dirname[static PATH_MAX]` would allow the proper compiler check. However that syntax is not much used because we often end up passing a pointer.